---
title: "combining study findings"
format: html
---

# combinding study findings

this scripts will take the p-values from differential expression and differential isoform expression

for joining p-values of the same tissue and expression we will use edgingtons method. it will be performed by:

`metap::sump(c(result1,results2)$p)`

For studies of different tissues or methods, we will use Fishers method, which prioritizes the smallest p-value, as we in the end want genes that are significant for any of the tested properties (tissue, expression, isoform ...)

`p_join_fisher = metap::sumlog(c(pvalue_deseq2,pvalue_dexseq))$p )`

## load DEseq2 results and DEXseq results

```{r}




#load deseq
load("/home/projects/exam_2025_22140_22150/group_2/data/GSE173955_deseq_result.Rdata")
load("/home/projects/exam_2025_22140_22150/group_2/data/GSE67333_deseq_result.Rdata")

#load dexseq data
load("/home/projects/exam_2025_22140_22150/group_2/data/GSE67333_DXD.Rdata")
GSE67333_DXD
```

## combinding p-values for same studies

### combinding DE

for studies performing the same test on the same tissue, we will combine the p-values using the edgingtons method.

```{r}
library(tidyverse)
# full_join datasets
deseq_pvalues <- inner_join(
  GSE173955_results |> 
  as.data.frame() |> 
    dplyr::select(pvalue) |> 
    rownames_to_column(var = "gene_ID"), 
  GSE67333_results |> 
  as.data.frame() |> 
    dplyr::select(pvalue) |> 
    rownames_to_column(var = "gene_ID"),
  by = c("gene_ID" = "gene_ID")) 
  


# perform p-value join using edgingtons method
deseq_signif_genes <- deseq_pvalues |> 
  na.omit() |> 
  rowwise() |> 
  mutate(joined_pvalue = metap::sump(c(pvalue.x,pvalue.y))$p ) |> 
  ungroup() |> 
  mutate(p_adj_deseq = p.adjust(joined_pvalue,method = "fdr")) |> 
  filter(p_adj_deseq < 0.05) |> 
  pull(gene_ID)
save(deseq_signif_genes,
     file = "/home/projects/exam_2025_22140_22150/group_2/data/dseq_signif_genes.Rdata")

```

## combining p-values for different studies

```{r}

```
